synthesizer/src/vm/finalize.rs

```diff
@@ -571,7 +571,7 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
             // Acquire the write lock on the process.
             // Note: Due to the highly-sensitive nature of processing all `finalize` calls,
             // we choose to acquire the write lock for the entire duration of this atomic batch.
-            let mut process = self.process.write();
+            let process = self.process.write();
```

Comment:
this isn't part of this PR, but this further proves that a write guard isn't really necessary here; a read one would suffice

Reply:
Want to revive this PR onto AleoNet? https://github.com/ProvableHQ/snarkVM/pull/19

Reply:
https://github.com/AleoNet/snarkVM/pull/2568